src/xunit.execution/Sdk/Frameworks/Runners/XunitTestClassRunner.cs

```diff
@@ -88,9 +93,12 @@ protected virtual void CreateClassFixture(Type fixtureType)
         async Task CreateClassFixtureAsync(Type fixtureType)
         {
             CreateClassFixture(fixtureType);
-            foreach(var uninitializedFixture in ClassFixtureMappings.Values.OfType<IAsyncLifetime>())
-                await Aggregator.RunAsync(uninitializedFixture.InitializeAsync);
-            
+            foreach (var uninitializedFixture in ClassFixtureMappings.Values.OfType<IAsyncLifetime>().ToList())
+                if (!InitializedAsyncFixtures.Contains(uninitializedFixture))
+                {
+                    InitializedAsyncFixtures.Add(uninitializedFixture);
+                    await Aggregator.RunAsync(uninitializedFixture.InitializeAsync);
```

Comment:
Should we be awaiting here or using a `Task.WhenAll` outside of the loop? Seems like things may still happen serially here?